### PR TITLE
Calendar style and prop update (original behaviour)

### PR DESCRIPTION
**Pull request by keithtxw**
**Changes**
Some adaptations from this [PR](https://github.com/LifeSG/react-design-system/pull/153)

- Update style to have navigation buttons at the top
- Introduce prop to control if the bordered style is rendered
- Update to `minDate` and `maxDate` props from `between`
- **Note: Original behaviour is retained. Update to be dealt with in another PR**

**TODO**
- Update the disabled behaviour as discussed

- [delete] branch

**Changelog entry**

- [BREAKING] Replace `between` to `minDate` and `maxDate`
- Add `styleType` prop to control if a border should appear or not
- Style updates

### CODE REVIEW

File: tests/utils/calendar-helper.spec.ts [hunk 1]
```diff
@@ -99,6 +99,87 @@ describe("CalendarHelper", () => {
...
+                date,
+                minDate,
+                maxDate
+            );
+
+            expect(isWithinRange).toBeTruthy();
+        });
+
+        it("should return true if the date is within range (inclusive check)", () => {
+            const date = dayjs("2023-02-01");
```
Comment (qroll) [inline tests/utils/calendar-helper.spec.ts:119]: can check for 2023-01-01 also (to assert min date date is correct)

File: src/shared/internal-calendar/internal-calendar.style.tsx [hunk 2]
```diff
@@ -100,20 +95,22 @@ export const OptionsOverlay = styled.div<OverlayStyleProps>`
     }}
 `;
 
-export const Header = styled.div`
+export const Header = styled.div<GeneralStyleProps>`
     display: flex;
     justify-content: space-between;
-    margin-bottom: 0.5rem;
-`;
+    margin: 0 0.5rem 1rem 0.5rem;
```
Comment (qroll) [inline src/shared/internal-calendar/internal-calendar.style.tsx:101]: i think Container can deal with the horizontal spacing which is 1.25rem for input and 2rem for standalone, header just needs to deal with the bottom margin

Comment (qroll) [reply src/shared/internal-calendar/internal-calendar.style.tsx:101]: actually some spacing and styles have deviated from the current Figma. better to address in another ticket.

**Review by qroll — CHANGES_REQUESTED** [review]

**Comment by qroll** [comment]
side note: to update date input and date range input which still use `between` to decide if manual date input is valid

File: src/shared/internal-calendar/internal-calendar.style.tsx [hunk 3]
```diff
@@ -191,6 +188,18 @@ export const HeaderArrowButton = styled(IconButton)`
     :active {
         background: transparent;
     }
+
+    ${(props) => {
+        if (props.$type === "standalone") {
+            return css`
+                ${ArrowLeft},
+                ${ArrowRight} {
+                    height: 1.125rem;
```
[the remark this answers is not in the document]
Comment (keithtxw) [reply src/shared/internal-calendar/internal-calendar.style.tsx:197]: Oh really? They unified the specs?

File: src/shared/internal-calendar/internal-calendar.style.tsx (hunk 2 above)
Comment (keithtxw) [reply src/shared/internal-calendar/internal-calendar.style.tsx:101]: will deal with it in the refactor to have CalendarWrapper

File: stories/storybook-common/api-table/markup-helpers.tsx [hunk 4]
```diff
@@ -0,0 +1,3 @@
+import React from "react";
+
+export const code = (content: string) => <code>{content}</code>;
```
Comment (keithtxw) [inline stories/storybook-common/api-table/markup-helpers.tsx:3]: I think we can put the `quote` function here too

**Review by qroll — APPROVED** [review]

**Pull request merged**